rclcpp_action/src/server.cpp

```diff
@@ -45,6 +45,7 @@ class ServerBaseImpl
 
   // Lock everything except user callbacks
   std::recursive_mutex reentrant_mutex_;
+  std::recursive_mutex goal_request_mutex_;
```

Comment:
i think this can be private member of `ServerBase`.

Reply:
Do you mean both mutexes or only `goal_request_mutex_`?